### PR TITLE
feat(resourcepolicies): support dataMover parameter in snapshot volume

### LINKED ISSUE #9829: Data mover type select through volume policy
Enhance the volume policy to support data mover type selection

### PR DESCRIPTION
**Pull request by chlins**
Thank you for contributing to Velero!

# Please add a summary of your change
This pull request adds support for selecting the data mover type (`velero-fs` or `velero-block`) in the volume policy snapshot action via a new `dataMover` parameter. It introduces constants and validation logic for this parameter, provides a method to retrieve it, and adds comprehensive unit tests to ensure correct behavior and validation.

**Feature: Data mover selection for snapshot actions**
- Added support for specifying the data mover type (`velero`, `velero-fs`, or `velero-block`) in the snapshot action's parameters using the new `dataMover` parameter. [[1]](diffhunk://#diff-b4a211bb0da7003e8ded25ae4e54e84bfbce60c8d5772d3ac7d494a7dea34461R1) [[2]](diffhunk://#diff-ec2c1bb35711f0fbca0e0bcc1338614c25a9bfd5727b9f0833c75f26ec425813R51-R72)
- Introduced the `GetDataMover` method to the `Action` struct to retrieve the configured data mover for snapshot actions.

**Validation and error handling**
- Implemented validation logic to ensure the `dataMover` parameter is only accepted for snapshot actions, must be a string, and must be one of the allowed values.

**Testing**
- Added unit tests for the new `GetDataMover` method to cover various edge cases.
- Extended validation tests to cover valid and invalid usages of the `dataMover` parameter in resource policies.
# Does your change fix a particular issue?

Fixes https://github.com/velero-io/velero/issues/9829

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [x] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: internal/resourcepolicies/resource_policies.go [hunk 1]
```diff
@@ -48,6 +48,28 @@ const (
...
+const (
+	// DataMoverParameter is the key of the action parameter that selects the data
+	// mover to be used for the matched volumes when the action type is snapshot.
+	DataMoverParameter = "dataMover"
+
+	// DataMoverVelero refers to the default built-in data mover. The default data
+	// mover may change among releases; currently it is the file system data mover.
+	DataMoverVelero = "velero"
+	// DataMoverVeleroFS refers to the Velero file system data mover.
+	DataMoverVeleroFS = "velero-fs"
```
Comment (Lyndon-Li) [inline internal/resourcepolicies/resource_policies.go]: These values have been defined in pkg/datamover/util.go, let's share those definitions.

"" and "velero" are both valid data mover types, they both mean the default built-in data mover that is "velero-fs" at present. So please add `DataMoverVelero = "velero"` into pkg/datamover/util.go as well.

File: internal/resourcepolicies/resource_policies.go [hunk 2]
```diff
@@ -56,6 +78,25 @@ type Action struct {
 	Parameters map[string]any `yaml:"parameters,omitempty"`
 }
 
+// GetDataMover returns the data mover configured in the snapshot action's
+// parameters. The second return value is false when the dataMover parameter is
+// not set, in which case callers should fall back to the per-backup DataMover
+// value. It only applies to the snapshot action type.
+func (a *Action) GetDataMover() (string, bool) {
+	if a == nil || a.Type != Snapshot || len(a.Parameters) == 0 {
```
Comment (Lyndon-Li) [inline internal/resourcepolicies/resource_policies.go]: For `a == nil` or `a.Type != Snapshot`, since data mover parameter is for snapshot action only, I think we should return error.
Otherwise, for `len(a.Parameters) == 0`, it means data mover type is not specified, then we should regard it as the default built in data mover. I suggest we add a `GetDefaultBuiltInDataMover` function in pkg/datamover/util.go and call it here.

For other validation or conversion failures, we should also return errors.

File: internal/resourcepolicies/resource_policies.go (hunk 1 above)
Comment (chlins) [reply internal/resourcepolicies/resource_policies.go]: In order to avoid import cycle, moved the constant into `pkg/util/datamover`.

File: internal/resourcepolicies/resource_policies.go (hunk 2 above)
Comment (chlins) [reply internal/resourcepolicies/resource_policies.go]: Updated.

**Review by Lyndon-Li — APPROVED** [review]

File: pkg/util/datamover/datamover.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+// Package datamover holds the shared data mover type identifiers and helpers.
+// It must remain a leaf package (stdlib-only imports) so it can be referenced
+// from anywhere in the codebase without introducing import cycles.
+package datamover
+
+const (
+	// DataMoverTypeVelero refers to the default built-in data mover. The default
+	// data mover may change among releases; see GetDefaultBuiltInDataMover.
+	DataMoverTypeVelero = "velero"
```
Comment (blackpiglet) [inline pkg/util/datamover/datamover.go:25]: Part of the this file conflicts with file. 
https://github.com/velero-io/velero/blob/main/pkg/datamover/util.go

Comment (chlins) [reply pkg/util/datamover/datamover.go:25]: The conflict part has been removed in this PR, we moved some definitions from the file you mentioned to the `pkg/util/datamover` to avoid import cycle.

**Review by blackpiglet — APPROVED** [review]
